### PR TITLE
tasks/workunit.py: when cloning, use --depth=1

### PR DESCRIPTION
**Pull request by dmick**
Help avoid killing git.ceph.com.  A depth 1 clone takes about
7 seconds, whereas a full one takes about 3:40 (much of it
waiting for the server to create a huge compressed pack)

Signed-off-by: Dan Mick <dan.mick@redhat.com>

### CODE REVIEW

**Comment by dmick** [comment]
Tested with:

```
interactive-on-error: true
roles: 
  - [mon.a, osd.0, client.0]
  - [mon.b, mon.c, osd.1, osd.2]
tasks:
  - install:
      branch: master
  - ceph:
  - workunit:
      clients:
        client.0: [cephtool]

```

File: qa/tasks/workunit.py [hunk 1]
```diff
@@ -321,6 +321,7 @@ def _run_tests(ctx, refspec, role, tests, env, subdir=None, timeout=None):
                 run.Raw('&&'),
                 'git',
                 'clone',
+                '--depth=1',
```
Comment (tchaikov) [inline qa/tasks/workunit.py]: we can also pass `--no-checkout,` to `git clone`, to save some time to check out the HEAD. we will run `git check` later one anyway.

Comment (dmick) [inline qa/tasks/workunit.py]: we could, but the time is totally dominated by the remote pack creation and transmission.  I can't observe any difference at all from --no-checkout.

**Comment by dmick** [comment]
I guess the failure is not surprising; this test hasn't run for 13 months:

http://pulpito.ceph.com/?suite=rados%3Asingleton%3Aall

**Comment by tchaikov** [comment]
this breaks all the workunit tests, if they are not running against master.

**Pull request merged**